File: app/src/main/java/afkt/project/ui/activity/ActivityResultCallbackActivity.java
package afkt.project.ui.activity;

import android.app.Activity;
import android.content.Intent;
import android.view.View;

import com.luck.picture.lib.PictureSelector;
import com.luck.picture.lib.config.PictureMimeType;
import com.luck.picture.lib.entity.LocalMedia;

import afkt.project.R;
import afkt.project.base.app.BaseActivity;
import afkt.project.databinding.ActivityActivityResultCallbackBinding;
import dev.other.picture.PictureSelectorUtils;
import dev.utils.app.ActivityUtils;
import dev.utils.app.AppUtils;
import dev.utils.app.toast.ToastTintUtils;

/**
 * detail: 跳转 Activity 回传 Callback
 * @author Ttt
 */
public class ActivityResultCallbackActivity
        extends BaseActivity<ActivityActivityResultCallbackBinding> {

    @Override
    public int baseLayoutId() {
        return R.layout.activity_activity_result_callback;
    }

    @Override
    public void initListener() {
        super.initListener();
        binding.vidAarcSelectBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                AppUtils.startActivityForResult(new ActivityUtils.ResultCallback() {
                    @Override
                    public boolean onStartActivityForResult(Activity activity) {
                        // 初始化图片配置
                        PictureSelectorUtils.PicConfig picConfig = new PictureSelectorUtils.PicConfig()
                                .setCompress(false).setMaxSelectNum(1).setCrop(false).setMimeType(PictureMimeType.ofImage())
                                .setCamera(true).setGif(false);
                        // 打开图片选择器
                        PictureSelectorUtils.openGallery(PictureSelector.create(activity), picConfig);
                        return true;
                    }

                    @Override
                    public void onActivityResult(
                            boolean result,
                            int resultCode,
                            Intent data
                    ) {
                        if (result && data != null) {
                            LocalMedia localMedia = PictureSelectorUtils.getSingleMedia(data);
                            // 获取图片地址
                            String imgPath = PictureSelectorUtils.getLocalMediaPath(localMedia, true);
                            // 提示
                            ToastTintUtils.success("选择了图片: " + imgPath);
                        } else {
                            showToast(false, "非成功操作");
                        }
                    }
                });
            }
        });
    }
}

File: app/src/main/java/afkt/project/ui/activity/ActivityResultCallbackActivity.kt
package afkt.project.ui.activity

import afkt.project.R
import afkt.project.base.app.BaseActivity
import afkt.project.base.config.RouterPath
import afkt.project.databinding.ActivityActivityResultCallbackBinding
import android.app.Activity
import android.content.Intent
import com.alibaba.android.arouter.facade.annotation.Route
import dev.engine.DevEngine
import dev.engine.media.MediaConfig
import dev.utils.app.ActivityUtils
import dev.utils.app.AppUtils
import dev.utils.app.toast.ToastTintUtils

/**
 * detail: 跳转 Activity 回传 Callback
 * @author Ttt
 */
@Route(path = RouterPath.ActivityResultCallbackActivity_PATH)
class ActivityResultCallbackActivity : BaseActivity<ActivityActivityResultCallbackBinding>() {

    override fun baseLayoutId(): Int = R.layout.activity_activity_result_callback

    override fun initListener() {
        super.initListener()
        binding.vidSelectBtn.setOnClickListener {
            AppUtils.startActivityForResult(object : ActivityUtils.ResultCallback {
                override fun onStartActivityForResult(activity: Activity): Boolean {
                    // 初始化图片配置
                    val config = MediaConfig()
                        .setCompress(false).setMaxSelectNum(1).setCrop(false)
                        .setMimeType(MediaConfig.MimeType.ofImage())
                        .setCamera(true).setGif(false)
                    // 打开图片选择器
                    DevEngine.getMedia()?.openGallery(activity, config)
                    return true
                }

                override fun onActivityResult(
                    result: Boolean,
                    resultCode: Int,
                    data: Intent?
                ) {
                    if (result && data != null) {
                        val imgPath = DevEngine.getMedia()?.getSingleSelectorPath(data, true)
                        // 提示
                        ToastTintUtils.success("选择了图片: $imgPath")
                    } else {
                        showToast(false, "非成功操作")
                    }
                }
            })
        }
    }
}